### PR TITLE
Validation testing

### PR DESCRIPTION
**Pull request by mirakst**
Now splits into train, validation and test sets at a 60, 20, 20 ratio (from one of our meetings). Is configurable, test size is inferred.
The validation step is performed after the training step in every epoch, and calculates loss for predictions with no backwards propagation (no training is performed, only measurement of model accuracy).
In practice, it is extremely similar to the test step (I think the code is probably like 90% the same), so could probably just call the test method. Thoughts?

### CODE REVIEW

**Review by damniko — APPROVED** [review]
LGTM

**Review by KasperHenningsen — APPROVED** [review]
LGTM ⛵️

**Review by AndersMariegaard — APPROVED** [review]
